app/controllers/notifications_controller.rb

```diff
@@ -39,6 +39,12 @@ def index
         changed = current_user.saw_notification_id(max_id)
       end
 
+      if !params.has_key?(:silent) && !@readonly_mode && guardian.can_see_review_queue?
```

Comment:
Are we able to move the guardian check into the defer queue too? That would eliminate one query for non staff user from the request and be placed into the background thread.

Reply:
done, although I wonder about concurrency safety here 🤔 like could the objects returned by the `guardian` and/or `current_user` methods be changed by another request by the time the deferred block executed? I vaguely remember we had a concurrency issue in the past that involved `Scheduler::Defer.later`

Reply:
Good point, to be safe you can always recreate the guardian in the thread.